python/cudf/cudf/core/frame.py

```diff
@@ -1817,6 +1817,11 @@ def sample(
                 n = int(round(self.shape[1] * frac))
 
         if axis is None or axis == 0 or axis == "index":
+            if n > 0 and self.shape[0] == 0:
+                raise ValueError(
+                    "a must be greater than 0 unless no samples are taken"
```

Comment:
What must be greater than zero? `a`? I don't see an `a` defined. Can you check this?

Reply:
Yeah so that's the error message Pandas throws in this case where `a` refers to the length of the given axis. I can change it to something a little more clear, but the tests check for exception message parity so I'd have to disable that in these instances if that's alright.

Reply:
Lets adjust the message to be as clear as possible and just make sure that we're throwing the same exception type. This was something like:
```
try:
    ...
except ValueError as e:
    ...
```
works as expected